drivers/i2c/i2c_gecko.c

```diff
@@ -75,8 +76,12 @@ static int i2c_gecko_configure(const struct device *dev, uint32_t dev_config_raw
 	i2cInit.master = 0;
 #endif
 
+	k_sem_take(&data->bus_mutex, K_FOREVER);
```

Comment:
the lock doesn't encompass data->dev_config mutation as it likely should

Reply:
Encompass from data->dev_config